staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -320,6 +318,26 @@ func (o *WaitOptions) RunWait() error {
 	ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), o.Timeout)
 	defer cancel()
 
+	if strings.ToLower(o.ForCondition) == "create" {
+		if err := wait.PollUntilContextTimeout(ctx, 500*time.Millisecond, o.Timeout, true, func(context.Context) (done bool, err error) {
```

Comment:
This code block will be executable in followup PRs?. Because `--for=create` currently returns immediately by L:196

Reply:
Not sure what you're asking, but my hope is to turn this block into pre-conditions, each function we support could have a pre-condition (currently on create will have one) and an actual condition one (all of them). 

Only in the long run, when we switch away from ResourceBuilder (or expose the necessary parts of resource parsing from there) we'll be able to use dynamic client more openly. So we need to make sure the preconditions is for now an implementation detail people should not rely :wink: